lib/engine/game/g_18_zoo/game.rb

```diff
@@ -1483,6 +704,40 @@ def share_price_updated(entity, revenue)
           stock_market.find_share_price(entity, :stay)
         end
 
+        def route_distance(route)
+          distance = route.visited_stops.sum(&:visit_cost)
+          return distance if @holes.empty?
+
+          distance + ((@holes & route.all_hexes).size == 2 ? 1 : 0)
+        end
+
+        def assign_hole(entity, target)
+          @holes << target
+
+          return unless entity.all_abilities.empty?
```

Comment:
Apart from this method lgtm.

Shouldn't the entity check it has the right ability?

Reply:
What do the 'holes' actually do?